core/src/labels/label.h

```diff
@@ -42,18 +42,26 @@ class Label {
     };
 
     struct Vertex {
-        Vertex(glm::vec2 pos, glm::vec2 uv, glm::vec3 extrude, uint32_t color, uint32_t stroke = 0)
-            : pos(pos), uv(uv), extrude(extrude), color(color), stroke(stroke) {}
-
-        glm::vec2 pos;
-        glm::vec2 uv;
-        glm::vec3 extrude;
+        Vertex(glm::vec2 pos, glm::vec2 uv, glm::vec2 extrude, uint32_t color, uint32_t stroke = 0)
+            : pos(pos * 4.f), uv(uv),
+              extrude(extrude * 256.f),
+              color(color), stroke(stroke) {}
+
+        glm::i16vec2 pos;
+        glm::i16vec2 uv;
+        glm::i16vec2 extrude;
         uint32_t color;
         uint32_t stroke;
         struct State {
-            glm::vec2 screenPos;
-            float alpha = 0.f;
-            float rotation = 0.f;
+            State() {}
+            State(glm::vec2 pos, float alpha, float rotation)
+                : screenPos(pos * 4.f),
+                  alpha(alpha * 32767.f),
+                  rotation(rotation * 4096.f) {}
```

Comment:
I didn't want to use defines in a header (as they should be undefined at the end to not leak to the outside). constexpr with anynomous namespace seem to be a good alternative for local defines generally.